wicker/core/datasets.py

```diff
@@ -1,30 +1,103 @@
 import abc
+import logging
 import os
 from functools import cached_property
 from multiprocessing.pool import ThreadPool
-from typing import Any, Dict, List, Optional, Tuple
+from typing import Any, Dict, List, Optional, Tuple, Union
 
 import boto3
 import pyarrow  # type: ignore
 import pyarrow.fs as pafs  # type: ignore
 import pyarrow.parquet as papq  # type: ignore
 import tqdm  # type: ignore
 
-from wicker.core.column_files import ColumnBytesFileCache, ColumnBytesFileLocationV1
+from wicker.core.column_files import (
+    ColumnBytesFileCache,
+    ColumnBytesFileLocationV1,
+    ColumnBytesFileReader,
+)
 from wicker.core.config import get_config  # type: ignore
 from wicker.core.definitions import DatasetDefinition, DatasetID, DatasetPartition
 from wicker.core.shuffle import ShuffleWorker
-from wicker.core.storage import S3DataStorage, S3PathFactory
+from wicker.core.storage import (
+    AbstractDataStorage,
+    FileSystemDataStorage,
+    S3DataStorage,
+    S3PathFactory,
+    WickerPathFactory,
+)
 from wicker.schema import dataloading, serialization
 from wicker.schema.schema import DatasetSchema
 
 # How long to wait before timing out on filelocks in seconds
 FILE_LOCK_TIMEOUT_SECONDS = 300
 
+logger = logging.getLogger(__name__)
+
 
 class AbstractDataset(abc.ABC):
     """Interface for a Map-style (non-streaming) Dataset"""
 
+    def __init__(
```

Comment:
Why an abstract class is instantiating?

Reply:
You wanted the logic consolidated

Reply:
@aalavian in Python (just like in other object-oriented languages) it is ok for abstract base classes to define constructors. They need to define at least one `@abc.abstractmethod` function in order to actually be abstract (which can be another function besides `__init__`).